docs/programming_guide/controllers/model_controller.rst

```diff
@@ -0,0 +1,247 @@
+================
+
+persistor
+---------
+The :func:`save_model<nvflare.app_common.workflows.model_controller.ModelController.save_model>` and :func:`load_model<nvflare.app_common.workflows.model_controller.ModelController.load_model>`
+functions utilize the configured persistor set in the ModelController ``persistor_id: str = "persistor"`` argument.
+
+For the JobAPI, the persistor with ``id = "persistor"`` will automatically be configured based on the type of the model sent to the server.
+
+.. code-block:: python
+
+  job.to(Net(), "server")
+
+The persistor can also be configured in ``config_fed_server.json`` in the components section.
+
+custom save & load
+------------------
+Users can also choose to instead create their own save and load functions rather than use a persistor.
+
+For example we can use PyTorch's save and load functions for the model parameters, and save the FLModel metadata with :mod:`FOBS<nvflare.fuel.utils.fobs>` separately:
+
+.. code-block:: python
+
+  import torch
+  from nvflare.fuel.utils import fobs
+
+  def save_model(self, model, filepath=""):
+        params = model.params
+        # PyTorch save
+        torch.save(params, filepath)
+
+        # save FLModel metadata
+        model.params = {}
+        fobs.dumpf(model, filepath + ".metadata")
+        model.params = params
+
+    def load_model(self, filepath=""):
+        # PyTorch load
+        params = torch.load(filepath)
+
+        # load FLModel metadata
+        model = fobs.loadf(filepath + ".metadata")
+        model.params = params
+        return model
+
+
+Note: for non-primitive data types such as ``torch.nn.Module`` (used for the initial PyTorch model), we must configure a corresponding FOBS decomposer for serialization and deserialization.
+Read more at :github_nvflare_link:`Flare Object Serializer (FOBS) <nvflare/fuel/utils/fobs/README.rst>`.
+
+.. code-block:: python
+
+  from nvflare.app_opt.pt.decomposers import TensorDecomposer
+
+  fobs.register(TensorDecomposer)
+
+
+Additional Functionalities
+==========================
+
+In some cases, more advanced FLARE-specific functionalities may be of use.
+
+The :mod:`BaseModelController<nvflare.app_common.workflows.base_model_controller>` class provides access to the engine ``self.engine`` and FLContext ``self.fl_ctx`` if needed.
+Functions such as ``get_component()`` and ``build_component()`` can be used to load or dynamically build components.
+
+Furthermore, the underlying :mod:`Controller<nvflare.apis.impl.controller>` class offers additional communication functions and task related utilities.
+Many of our pre-existing workflows are based on this lower-level Controller API.
+For more details refer to the :ref:`controllers` section.
+
+Configuration
+=============
+
+For the JobAPI, define the controller and send it to the server.
+
+.. code-block:: python
+
+  controller = FedAvg(
+      num_clients=n_clients,
+      num_rounds=num_rounds,
+  )
+  job.to(controller, "server")
```

Comment:
Not sure if we want this section in "ModelController" though.
This should be a generic thing for all controllers?
So maybe we should put this in some workflow configuration doc that is in "user_guide" or something?